### PR TITLE
readme: add Advanced Usage Patterns section

### LINKED ISSUE #153: Document 'Advanced Usage'
An `Advanced Usage` section would be a nice addition to the Readme. It could contain;

- Piping to `s5cmd run`: a clever usage is documented at @joshuarobinson's [article](https://medium.com/@joshua_robinson/s5cmd-hits-v1-0-and-intro-to-advanced-usage-37ad02f7e895) (Thanks for the article! Again! 💯 )
- Suggest using wildcards if there are many objects to be deleted and they share a common pattern. For example, consider these objects;
```
s3://bucket/prefix/2020/03/object1.gz
s3://bucket/prefix/2020/04/object1.gz
````

Instead of removing like
```
s3://bucket/prefix/2020/03/object1.gz
s3://bucket/prefix/2020/04/object1.gz
````

Suggest:
```
s3://bucket/prefix/2020/0*/object*.gz
````

Using wildcard for removing many objects dramatically improves execution times (single request per 1000 objects) since s5cmd utilizes S3 Batch API.

### PR DESCRIPTION
**Pull request by fbarotov**
Resolves #153

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -329,6 +329,56 @@ of concurrent systems easy and make full utilization of multi-core processors.
 - *Parallelization.* `s5cmd` starts out with concurrent worker pools and parallelizes
 workloads as much as possible while trying to achieve maximum throughput.
 
+# Advanced Usage
+
+Some of the advanced usage patterns provided below are inspired by the following [article](https://medium.com/@joshua_robinson/s5cmd-hits-v1-0-and-intro-to-advanced-usage-37ad02f7e895) (thank you! [@joshuarobinson](https://github.com/joshuarobinson))
+
+## Integrate s5cmd operations with Unix commands
+
+    s5cmd ls s3://bucket/pre/ | sort
```
Comment (igungor) [inline README.md:338]: what do you think of adding a more useful example; peeking at the schema of avocado 🥑 prices dataset. You don't have to download the whole file, just fetch the necessary few bytes.

```
~  Δ s5cmd cat s3://bucket/avocado.csv.gz | gunzip | xsv slice --len 5 | xsv table
    Date        AveragePrice  Total Volume  4046     4225       4770   Total Bags  Small Bags  Large Bags  XLarge Bags  type          year  region
0   2015-12-27  1.33          64236.62      1036.74  54454.85   48.16  8696.87     8603.62     93.25       0.0          conventional  2015  Albany
1   2015-12-20  1.35          54876.98      674.28   44638.81   58.33  9505.56     9408.07     97.49       0.0          conventional  2015  Albany
2   2015-12-13  0.93          118220.22     794.7    109149.67  130.5  8145.35     8042.21     103.14      0.0          conventional  2015  Albany
3   2015-12-06  1.08          78992.15      1132.0   71976.41   72.58  5811.16     5677.4      133.76      0.0          conventional  2015  Albany
4   2015-11-29  1.28          51039.6       941.48   43838.39   75.78  6183.95     5986.26     197.69      0.0          conventional  2015  Albany
```

Comment (fbarotov) [reply README.md:338]: thanks for the feedback and help. done.

File: README.md [hunk 2]
```diff
@@ -329,6 +329,67 @@ of concurrent systems easy and make full utilization of multi-core processors.
...
+
+    s5cmd cat s3://bucket/file.txt | grep something
```
Comment (igungor) [inline README.md:340]: We already have one example for `cat`. Let's remove this one.

File: README.md [hunk 3]
```diff
@@ -329,6 +329,67 @@ of concurrent systems easy and make full utilization of multi-core processors.
 - *Parallelization.* `s5cmd` starts out with concurrent worker pools and parallelizes
 workloads as much as possible while trying to achieve maximum throughput.
 
+# Advanced Usage
+
+Some of the advanced usage patterns provided below are inspired by the following [article](https://medium.com/@joshua_robinson/s5cmd-hits-v1-0-and-intro-to-advanced-usage-37ad02f7e895) (thank you! [@joshuarobinson](https://github.com/joshuarobinson))
+
+## Integrate s5cmd operations with Unix commands
+
+    s5cmd ls s3://bucket/pre/ | sort
```
Comment (igungor) [inline README.md:338]: Could you add a few words about what it does? Also the output would be nice.

File: README.md [hunk 4]
```diff
@@ -334,10 +334,15 @@ workloads as much as possible while trying to achieve maximum throughput.
 Some of the advanced usage patterns provided below are inspired by the following [article](https://medium.com/@joshua_robinson/s5cmd-hits-v1-0-and-intro-to-advanced-usage-37ad02f7e895) (thank you! [@joshuarobinson](https://github.com/joshuarobinson))
 
 ## Integrate s5cmd operations with Unix commands
-
-    s5cmd ls s3://bucket/pre/ | sort
-
-    s5cmd cat s3://bucket/file.txt | grep something
+Assume we have a set of objects on S3, and we would like to list them in sorted fashion according to object names.
```
Comment (igungor) [inline README.md:337]: nice 👍

**Comment by igungor** [comment]
Thanks @fbarotov 🥇

**Comment by fbarotov** [comment]
> Thanks @fbarotov 🥇

thank you for your help and guidance.

**Pull request merged**